### ISSUE TITLE
Bad text rendering

### ISSUE THREAD

**Issue by timvandermeij** [mozilla/pdf.js#6115]
PDF: http://www.kennemerlyceum.nl/attachments/article/4/Gymnasiumfolder.pdf
OS: Windows 7 x64
FF version: 38.0.5
PDF.js version: latest master

This seems to have regressed, either in PDF.js or in Firefox. This used to render fine, but now the text on the second page renders horribly. Perhaps it's related to the fact that the first page won't render completely if you scroll down immediately after opening the file (while the first page is still loading). Then the console reads:

`InvalidStateError: An attempt was made to use an object that is not, or is no longer, usable`

Screenshot (click to increase size and to see the problem):
![img](https://cloud.githubusercontent.com/assets/1993262/8149877/c8e55862-12d4-11e5-8209-79bab792855f.png)

**Comment by Snuffleupagus** [comment]
I can reproduce this in e.g. Firefox 38 (the current release), and in all never versions as well (Beta, Aurora, Nightly), using the latest PDF.js version.
However, this is almost certainly a Firefox regression, since it looks fine in PDF.js version `1.1.207` when testing in Firefox 31 (i.e. the ESR version). See the following screen-shot:

![esr31](https://cloud.githubusercontent.com/assets/2692120/8150421/3c11a1ce-12e9-11e5-896d-a1065f099d41.png)

**Comment by timvandermeij** [comment]
Marking as upstream issue. I hope someone can report it upstream once it has been triaged.

**Comment by yurydelendik** [comment]
mozregression bisect results:

```
Last good revision: 34a66aaaca81 (2015-02-05)
First bad revision: 7c5f187b65bf (2015-02-06)
Pushlog:
https://hg.mozilla.org/mozilla-central/pushloghtml?fromchange=34a66aaaca81&tochange=7c5f187b65bf


Last good revision: 0c2f7434c325
First bad revision: 351d130b72ef
Pushlog:
https://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?fromchange=0c2f7434c325&tochange=351d130b72ef
```

**Comment by timvandermeij** [comment]
Perhaps that part of the commit message indicates that it might a problem?

<pre>
Note that this fixes most situation, but in the case of doing custom blending to an area of a surface which is not opaque while having a complex clip pushed this will still lead to some artifacts. I haven't seen this be a problem in practice though.
</pre>

(from https://hg.mozilla.org/integration/mozilla-inbound/rev/0678a9a04d59)

Is there already an upstream issue for that and if so, would you mind filing one?

**Comment by yurydelendik** [comment]
Perhaps, we will need a reduced test case that is using canvas though to expedite the resolution.

**Comment by yurydelendik** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1175145

**Comment by timvandermeij** [comment]
I don't have Windows anymore to verify this, but this is fixed for me on Arch Linux using Firefox 45.0.2.

@Snuffleupagus Could you check if you can still reproduce this on Windows using the latest Firefox version? The upstream bug is not closed yet, but https://bugzilla.mozilla.org/show_bug.cgi?id=1175145#c4 might indicate that is actually is fixed.

**Comment by Snuffleupagus** [comment]
This WFM now, using Windows 7 (64-bit), and Firefox Nightly 49 (with HWA on):
```
Mozilla/5.0 (Windows NT 6.1; WOW64; rv:49.0) Gecko/20100101 Firefox/49.0 ID:20160504043118 CSet: 369a5ee3a2880a4a98df3a00bf3db8d8f36b181b
```
This is an excerpt from the `about:support` page:
```
Graphics
Features
Compositing	Direct3D 11
Asynchronous Pan/Zoom	none
WebGL Renderer	Google Inc. -- ANGLE (AMD Radeon HD 5800 Series Direct3D11 vs_5_0 ps_5_0)
Hardware H264 Decoding	No; Hardware video decoding disabled or blacklisted
Direct2D	true
DirectWrite	true (6.2.9200.17568)
GPU #1
Active	Yes
Description	AMD Radeon HD 5800 Series
Vendor ID	0x1002
Device ID	0x6899
Driver Version	13.152.0.0
Driver Date	8-30-2013
Drivers	aticfx64 aticfx64 aticfx64 aticfx32 aticfx32 aticfx32 atiumd64 atidxx64 atidxx64 atiumdag atidxx32 atidxx32 atiumdva atiumd6a atitmm64
Subsys ID	29701682
RAM	1024
Diagnostics
ClearType Parameters	Gamma: 2200 Pixel Structure: R ClearType Level: 100 Enhanced Contrast: 50
AzureCanvasAccelerated	0
AzureCanvasBackend	direct2d 1.1
AzureContentBackend	direct2d 1.1
AzureFallbackCanvasBackend	cairo
ClearType Parameters	Gamma: 2200 Pixel Structure: R ClearType Level: 100 Enhanced Contrast: 50 
```

Closing as works for me.
